distributed/scheduler.py

```diff
@@ -5894,7 +5899,7 @@ def transition(self, key, finish, *args, **kwargs):
             if ts._state == "forgotten" and ts._group._name in self.task_groups:
                 # Remove TaskGroup if all tasks are in the forgotten state
                 tg: TaskGroup = ts._group
-                if not any([tg._states.get(s) for s in ALL_TASK_STATES]):
+                if not (tg._states.keys() & ALL_TASK_STATES):
```

Comment:
As key views are `Set`-like objects, things like intersections can be computed with `set`s. More details in [this SO answer]( https://stackoverflow.com/a/18554039/3877089 ) along with [a link to the Python docs]( https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects ) where they note this:

> Keys views are set-like since their entries are unique and hashable. If all values are hashable, so that `(key, value)` pairs are unique and hashable, then the items view is also set-like. (Values views are not treated as set-like since the entries are generally not unique.) For set-like views, all of the operations defined for the abstract base class [`collections.abc.Set`]( https://docs.python.org/3/library/collections.abc.html#collections.abc.Set ) are available (for example, `==`, `<`, or `^`).

Reply:
In addition to simplifying the code a bit, this ends up being a bit more performant

```python
In [1]: d1 = {i: i for i in range(1_000)}
   ...: s2 = {i for i in range(1_000, 2_000)}

In [2]: %timeit any([d1.get(k) for k in s2])
177 µs ± 456 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [3]: %timeit d1.keys() & s2
15.8 µs ± 157 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Plus Cython can replace `&` with a quick C call to `PyNumber_And`.